pkg/resource/deploy/step.go

```diff
@@ -925,25 +926,40 @@ func NewRefreshStep(deployment *Deployment, old *resource.State, done chan<- boo
 	}
 }
 
-func (s *RefreshStep) Op() display.StepOp      { return OpRefresh }
-func (s *RefreshStep) Deployment() *Deployment { return s.deployment }
-func (s *RefreshStep) Type() tokens.Type       { return s.old.Type }
-func (s *RefreshStep) Provider() string        { return s.old.Provider }
-func (s *RefreshStep) URN() resource.URN       { return s.old.URN }
-func (s *RefreshStep) Old() *resource.State    { return s.old }
-func (s *RefreshStep) New() *resource.State    { return s.new }
-func (s *RefreshStep) Res() *resource.State    { return s.old }
-func (s *RefreshStep) Logical() bool           { return false }
+func (s *RefreshStep) Op() display.StepOp                           { return OpRefresh }
+func (s *RefreshStep) Deployment() *Deployment                      { return s.deployment }
+func (s *RefreshStep) Type() tokens.Type                            { return s.old.Type }
+func (s *RefreshStep) Provider() string                             { return s.old.Provider }
+func (s *RefreshStep) URN() resource.URN                            { return s.old.URN }
+func (s *RefreshStep) Old() *resource.State                         { return s.old }
+func (s *RefreshStep) New() *resource.State                         { return s.new }
+func (s *RefreshStep) Res() *resource.State                         { return s.old }
+func (s *RefreshStep) Logical() bool                                { return false }
+func (s *RefreshStep) Diffs() []resource.PropertyKey                { return s.diff.ChangedKeys }
+func (s *RefreshStep) DetailedDiff() map[string]plugin.PropertyDiff { return s.diff.DetailedDiff }
 
 // ResultOp returns the operation that corresponds to the change to this resource after reading its current state, if
 // any.
 func (s *RefreshStep) ResultOp() display.StepOp {
 	if s.new == nil {
 		return OpDelete
 	}
-	if s.new == s.old || s.old.Outputs.Diff(s.new.Outputs) == nil {
-		return OpSame
+
+	// Prior to us introducing refresh diffs against desired state, we only
+	// diffed outputs. For now, we support reverting to this behaviour for
+	// users that rely on it. We only need to make this check here as all
+	// downstream consumers will behave correctly based on whether s.diff is
+	// set or not.
```

Comment:
```suggestion
	// users that rely on it.
```

The rest of the sentence belongs in a commit/PR message rather than a code comment imo, since it gets confusing to tie together what other changes could possibly need to be made if unless this is viewed in the context of the PR.